File: guess-number/guess.dart
import "dart:core";
import "dart:io";
import "dart:math";

void main() {
    var n = 100;
    guess(n);
}

void guess(int n) {
    var random = Random(DateTime.now().millisecondsSinceEpoch);
    var number = random.nextInt(n) + 1;
    var guess;
    var attempt = 0;

    print('Guess a number between 1 and $n');

    while (attempt < 10) {
        attempt++;

        stdout.write('Please type in your guess: ');
        guess = int.tryParse(stdin.readLineSync().toString());
        if (guess == null) {
            continue;
        }

        if (guess.compareTo(number) < 0) {
            print('Higher number please');
        } else if (guess.compareTo(number) > 0) {
            print('Lower number please');
        } else {
            print('Bingo! You guessed the number in $attempt attempts');
            return;
        }
    }

    print('Oops! You lose after $attempt attempts');
}

File: guess-number/guess.js
function guess(n) {
    const number = Math.floor(Math.random() * n) + 1;
    let attempt = 0, guess;

    console.log(`Guess a number between 1 and ${n}`);

    const rl = require('readline').createInterface({
        input: process.stdin,
        output: process.stdout,
        prompt: 'Please type in your guess: '
    });

    rl.prompt();

    rl.on('line', (line) => {
        attempt++;

        guess = parseInt(line);
        if (guess < number && attempt < 10) {
            console.log('Higher number please');
        } else if (guess > number && attempt < 10) {
            console.log('Lower number please');
        } else if (guess === number) {
            console.log(`Bingo! You guessed the number in ${attempt} attempts`);
            rl.close();
            return;
        }

        if (attempt === 10) {
            console.log(`Oops! You lose after ${attempt} attempts`);
            rl.close();
            return;
        }
        rl.prompt();
    });
}

const n = 100;
guess(n);
